libs/react-helpers/src/hooks/use-data-provider.ts

```diff
@@ -10,17 +10,33 @@ import type { Subscribe } from '../lib/generic-data-provider';
  * @param variables optional
  * @returns state: data, loading, error, methods: flush (pass updated data to update function without delta), restart: () => void}};
  */
-export function useDataProvider<Data, Delta>(
-  dataProvider: Subscribe<Data, Delta>,
-  update?: (delta: Delta) => boolean,
-  variables?: OperationVariables
-) {
+export function useDataProvider<Data, Delta>({
+  dataProvider,
+  update,
+  insert,
+  variables,
+}: {
+  dataProvider: Subscribe<Data, Delta>;
+  update?: ({ delta, data }: { delta: Delta; data: Data }) => boolean;
+  insert?: ({
+    insertData,
+    data,
+    totalCount,
+  }: {
+    insertData: Data;
```

Comment:
nit (non-blocking): insertData sounds more like a function name. if this is actually data, i'd go with something `dataToInsert`, `insertionData` or `newData`. not super important, and i know we can hover over anything in our IDE and look at their type signature.

Reply:
Good point. I renamed it to `insertionData`